src/views/News/index.js

```diff
@@ -43,465 +70,556 @@ const styles = {
+        <NewsList
+          news={news}
+          personalUnapprovedNews={personalUnapprovedNews}
+          currentUsername={currentUsername}
+          handleNewsItemEdit={handleNewsItemEdit}
+          handleNewsItemDelete={handleNewsItemDelete}
+        />
       );
+    }
 
-      let news;
-      // If the user is online
-      if (
-        networkStatus === 'online' ||
-        (networkStatus === 'offline' && this.props.authentication)
-      ) {
-        news = (
-          <>
-            {/* Button to Create Posting */}
-            <Fab
-              variant="extended"
-              color="primary"
-              onClick={this.handlePostClick.bind(this)}
-              style={styles.fab}
+    let newsJSX;
+
+    // If the user is online
+    if (isOnline) {
+      newsJSX = (
+        <>
+          {/* Button to Create Posting */}
+          <Fab variant="extended" color="primary" onClick={handlePostClick} style={styles.fab}>
+            <PostAddIcon />
+            Post Listing
+          </Fab>
+
+          <Grid container justify="center">
+            {/* Search */}
+            <Grid item xs={12} md={12} lg={8}>
+              <Grid
+                container
+                alignItems="baseline"
+                justify="center"
+                style={styles.searchBar}
+                spacing={5}
+              >
+                <Grid item xs={10} sm={8} md={8} lg={6}>
+                  <TextField
+                    id="search"
+                    label="Search news"
+                    variant="filled"
+                    value={search}
+                    onChange={(event) => {
+                      setSearch(event.target.value);
+                    }}
+                    margin="normal"
+                    fullWidth
+                  />
+                </Grid>
+              </Grid>
+            </Grid>
+
+            {/* NOTE: leaving helper text for now in case
+            that is better than disabling submit button */}
+            {/* Create Posting */}
+            <GordonDialogBox
+              open={openPostActivity}
+              title=" Post on Student News "
+              buttonClicked={currentlyEditing ? handleUpdate : handleSubmit}
+              buttonName={currentlyEditing ? 'Update' : 'Submit'}
+              isButtonDisabled={submitButtonDisabled}
+              cancelButtonClicked={handleWindowClose}
+              cancelButtonName="Cancel"
             >
-              <PostAddIcon />
-              Post Listing
-            </Fab>
-
-            <Grid container justifyContent="center">
-              {/* Search */}
-              <Grid item xs={12} md={12} lg={8}>
-                <Grid
-                  container
-                  alignItems="baseline"
-                  justifyContent="center"
-                  style={styles.searchBar}
-                  spacing={5}
-                >
-                  <Grid item xs={10} sm={8} md={8} lg={6}>
+              <DialogContent>
```

Comment:
The `GordonDialogBox` already has a `DialogContent` wrapper tag, so I don't think this one is accomplishing much anymore.